Src/GeneratedTests/Discovery.Tests/DiscoveryTest.cs

```diff
@@ -0,0 +1,76 @@
+using Google.Apis.Http;
+using Google.Apis.Services;
+using NUnit.Framework;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Threading;
+using System.Collections.Generic;
+using System.Text;
+using System.Reflection;
+using System.IO;
+
+namespace Google.Apis.Discovery.v1.Tests
+{
+    [TestFixture]
+    public class DiscoveryTest
+    {
+        class FakeHttpMessageHandler : ConfigurableMessageHandler
+        {
+            public FakeHttpMessageHandler(Dictionary<string, byte[]> data) : base(new HttpClientHandler())
+            {
+                _data = data;
+            }
+
+            private Dictionary<string, byte[]> _data;
+
+            protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                var result = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+                {
+                    Content = new ByteArrayContent(_data[request.RequestUri.AbsolutePath])
+                };
+                return Task.FromResult(result);
+            }
+        }
+
+        class FakeHttpClientFactory : IHttpClientFactory
+        {
+            public FakeHttpClientFactory(Dictionary<string, byte[]> data)
+            {
+                _data = data;
+            }
+
+            private Dictionary<string, byte[]> _data;
+
+            public ConfigurableHttpClient CreateHttpClient(CreateHttpClientArgs args)
+            {
+                return new ConfigurableHttpClient(new FakeHttpMessageHandler(_data));
+            }
+        }
+
+        [Test]
+        public async Task TestUrlShortenerRef()
+        {
+            // Test that the "$ref" names as used by the discovery API are treated as normal names by the JSON deserializer.
+            // This is the first test of a generated library.
+            var stream = GetType().GetTypeInfo().Assembly.GetManifestResourceStream("Discovery.Tests.resources.urlshortener.json");
+            string sUrlShortener;
```

Comment:
This seems to go via a string for no obvious reason - we're basically just trying to read the whole of a resource and return it as a byte array, yes? Using MemoryStream and Stream.CopyTo is probably a cleaner approach here.